src/state/products.js

```diff
@@ -39,6 +39,57 @@ export const extractedProductShape = pt.shape({
   date: pt.string.isRequired,
 });
 
+const DATA_URI_PATHS = [
+  'image/gif;',
+  'image/png;',
+  'image/jpeg;',
+  'image/svg+xml;',
+  'image/webp;',
+];
+
+const RESOURCE_VALIDATION_DEFAULTS = {
+  isValidResource(value) {
+    try {
+      const url = new URL(value);
+      if (!['https:', 'http:'].includes(url.protocol)) {
+        return false;
+      }
+    } catch (err) {
+      return false;
+    }
+    return true;
+  },
+};
+
+const RESOURCE_VALIDATION = {
+  url: {
+    ...RESOURCE_VALIDATION_DEFAULTS,
+  },
+  image: {
+    ...RESOURCE_VALIDATION_DEFAULTS,
+    isValidResource(value) {
+      try {
+        const url = new URL(value);
+        const {protocol} = url;
+        if (!['https:', 'http:', 'data:'].includes(protocol)) {
+          return false;
+        }
+        if (protocol === 'data:') {
+          for (const path of DATA_URI_PATHS) {
+            if (url.pathname.startsWith(path)) {
+              return true;
+            }
+          }
+          return false;
```

Comment:
It's inside the `if (protocol === 'data:')` block, so no it wouldn't; it would only return `false` if the path for the data URI did not start with any of the strings in the `DATA_URI_PATHS` array.